
diff --git a/python/mach/mach/command_util.py b/python/mach/mach/command_util.py
--- a/python/mach/mach/command_util.py
+++ b/python/mach/mach/command_util.py
@@ -292,17 +292,17 @@ class DecoratorVisitor(ast.NodeVisitor):
         ]
 
         relevant_kwargs = ["command", "subcommand", "virtualenv_name"]
 
         for decorator in decorators:
             kwarg_dict = {}
 
             for name, arg in zip(["command", "subcommand"], decorator.args):
-                kwarg_dict[name] = arg.s
+                kwarg_dict[name] = arg.value
 
             for keyword in decorator.keywords:
                 if keyword.arg not in relevant_kwargs:
                     # We only care about these 3 kwargs, so we can safely skip the rest
                     continue
 
                 kwarg_dict[keyword.arg] = getattr(keyword.value, "s", "")
 




