diff -ru PyTrie-0.4.0.orig/pytrie.py PyTrie-0.4.0/pytrie.py
--- PyTrie-0.4.0.orig/pytrie.py	2025-12-25 11:00:21.199518391 +0100
+++ PyTrie-0.4.0/pytrie.py	2025-12-25 11:00:26.463479479 +0100
@@ -397,7 +397,7 @@
         self._root.children.clear()
 
     def copy(self):
-        clone = copy(super(Trie, self))
+        clone = copy(self)
         clone._root = copy(self._root)  # pylint: disable=protected-access
         return clone
 
