diff --git a/testing/mozbase/mozdebug/mozdebug/mozdebug.py b/testing/mozbase/mozdebug/mozdebug/mozdebug.py
index beecc2c..d8d0b85 100755
--- a/testing/mozbase/mozdebug/mozdebug/mozdebug.py
+++ b/testing/mozbase/mozdebug/mozdebug/mozdebug.py
@@ -6,9 +6,9 @@
 
 import json
 import os
+import shutil
 import sys
 from collections import namedtuple
-from distutils.spawn import find_executable
 from subprocess import check_output
 
 import mozinfo
@@ -108,7 +108,7 @@ def get_debugger_path(debugger):
             if path:
                 return path
         except Exception:
-            # Just default to find_executable instead.
+            # Just default to shutil.which instead.
             pass
 
     if mozinfo.os == "win" and debugger == "devenv.exe":
@@ -122,10 +122,10 @@ def get_debugger_path(debugger):
                 vsinfo[0]["installationPath"], "Common7", "IDE", "devenv.exe"
             )
         except Exception:
-            # Just default to find_executable instead.
+            # Just default to shutil.which instead.
             pass
 
-    return find_executable(debugger)
+    return shutil.which(debugger)
 
 
 def get_debugger_info(debugger, debuggerArgs=None, debuggerInteractive=False):
diff --git a/testing/mozbase/mozgeckoprofiler/mozgeckoprofiler/symbolication.py b/testing/mozbase/mozgeckoprofiler/mozgeckoprofiler/symbolication.py
index ecec5c1..7a7ff15 100644
--- a/testing/mozbase/mozgeckoprofiler/mozgeckoprofiler/symbolication.py
+++ b/testing/mozbase/mozgeckoprofiler/mozgeckoprofiler/symbolication.py
@@ -5,9 +5,9 @@ import hashlib
 import http.client
 import os
 import platform
+import shutil
 import subprocess
 import zipfile
-from distutils import spawn
 
 import six
 from mozlog import get_proxy_logger
@@ -95,7 +95,7 @@ class OSXSymbolDumper:
 
 class LinuxSymbolDumper:
     def __init__(self):
-        self.nm = spawn.find_executable("nm")
+        self.nm = spawn.shutil.which("nm")
         if not self.nm:
             raise SymbolError("Could not find nm, necessary for symbol dumping")
 
diff --git a/testing/mozbase/moznetwork/tests/test_moznetwork.py b/testing/mozbase/moznetwork/tests/test_moznetwork.py
index 8a1ee31..5bde9dc 100644
--- a/testing/mozbase/moznetwork/tests/test_moznetwork.py
+++ b/testing/mozbase/moznetwork/tests/test_moznetwork.py
@@ -5,12 +5,12 @@ Unit-Tests for moznetwork
 
 import re
 import subprocess
-from distutils.spawn import find_executable
 from unittest import mock
 
 import mozinfo
 import moznetwork
 import mozunit
+import shutil
 import pytest
 
 
@@ -37,7 +37,7 @@ def ip_addresses():
 
     cmd = None
     for command in commands:
-        if find_executable(command[0]):
+        if shutil.which(command[0]):
             cmd = command
             break
     else:
diff --git a/testing/mozbase/mozrunner/mozrunner/application.py b/testing/mozbase/mozrunner/mozrunner/application.py
index d1dd91e..d33e29b 100644
--- a/testing/mozbase/mozrunner/mozrunner/application.py
+++ b/testing/mozbase/mozrunner/mozrunner/application.py
@@ -4,8 +4,8 @@
 
 import os
 import posixpath
+import shutil
 from abc import ABCMeta, abstractmethod
-from distutils.spawn import find_executable
 
 import six
 from mozdevice import ADBDeviceFactory
@@ -51,7 +51,7 @@ class RemoteContext(object):
     @property
     def bindir(self):
         if self._bindir is None:
-            paths = [find_executable("emulator")]
+            paths = [shutil.which("emulator")]
             paths = [p for p in paths if p is not None if os.path.isfile(p)]
             if not paths:
                 self._bindir = ""
@@ -88,7 +88,7 @@ class RemoteContext(object):
             paths.insert(0, os.path.abspath(self.bindir))
             os.environ["PATH"] = os.pathsep.join(paths)
 
-        return find_executable(binary)
+        return shutil.which(binary)
 
     @abstractmethod
     def stop_application(self):
