diff --git a/meson.build b/meson.build
index c954a644c..e35eb0e97 100644
--- a/meson.build
+++ b/meson.build
@@ -279,10 +279,6 @@ endforeach
 cdata.set('HAVE_PIDFD_OPEN',
           cc.get_define('SYS_pidfd_open', prefix: '#include <sys/syscall.h>') != '')
 
-systemd_dep = dependency('libsystemd', required: get_option('libsystemd'))
-summary({'libsystemd': systemd_dep.found()}, bool_yn: true)
-cdata.set('HAVE_SYSTEMD', systemd_dep.found())
-
 logind_dep = dependency(get_option('logind-provider'), required: get_option('logind'))
 summary({'logind': logind_dep.found()}, bool_yn: true)
 cdata.set('HAVE_LOGIND', logind_dep.found())
