From b78878d89f543b86e377f716cc91036c73e7fe6e Mon Sep 17 00:00:00 2001
From: Dudemanguy <random342@airmail.cc>
Date: Tue, 8 Jun 2021 09:05:18 -0500
Subject: [PATCH] build: make systemd optional

---
 meson.build       | 6 +++---
 meson_options.txt | 5 +++++
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 7b29bfe..0260de1 100644
--- a/meson.build
+++ b/meson.build
@@ -15,7 +15,7 @@ glib_dep = dependency('glib-2.0', version: '>= 2.68')
 gio_dep = dependency('gio-2.0')
 gio_unix_dep = dependency('gio-unix-2.0')
 pipewire_dep = dependency('libpipewire-0.3', version: '>= 0.3.0')
-systemd_dep = dependency('systemd')
+systemd_dep = dependency('systemd', required: get_option('systemd'))
 libsecret_dep = dependency('libsecret-1')
 libnotify_dep = dependency('libnotify')
 
@@ -63,11 +63,11 @@ configure_file(input: 'config.h.meson',
 configinc = include_directories('.')
 
 servicedir = get_option('systemd_user_unit_dir')
-if servicedir == ''
+if servicedir == '' and systemd_dep.found()
   servicedir = systemd_dep.get_pkgconfig_variable('systemduserunitdir')
 endif
 
-if servicedir == ''
+if servicedir == '' and systemd_dep.found()
   error('Couldn\'t determine systemd user unit service directory')
 endif
 
diff --git a/meson_options.txt b/meson_options.txt
index ca2908b..3527588 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -8,6 +8,11 @@ option('vnc',
        value: true,
        description: 'Enable the VNC backend')
 
+option('systemd',
+       type: 'boolean',
+       value: true,
+       description: 'Enable systemd support')
+
 option('systemd_user_unit_dir',
        type: 'string',
        value: '',
-- 
2.32.0

