From 7f7948c077551c0cd76001ccf1538128d9c989b3 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Sat, 28 Aug 2021 19:16:56 -0500
Subject: [PATCH] format security

---
 zip.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/zip.c b/zip.c
index 439821f..91ad13b 100644
--- a/zip.c
+++ b/zip.c
@@ -1028,7 +1028,7 @@ local void help_extended()
 
   for (i = 0; i < sizeof(text)/sizeof(char *); i++)
   {
-    printf(text[i]);
+    printf("%s", text[i]);
     putchar('\n');
   }
 #ifdef DOS
@@ -1225,7 +1225,7 @@ local void version_info()
             CR_MAJORVER, CR_MINORVER, CR_BETA_VER, CR_VERSION_DATE);
   for (i = 0; i < sizeof(cryptnote)/sizeof(char *); i++)
   {
-    printf(cryptnote[i]);
+    printf("%s", cryptnote[i]);
     putchar('\n');
   }
   ++i;  /* crypt support means there IS at least one compilation option */
-- 
2.33.0

