From 33e6918c12221c59615568ae82a2b13ea272d9e5 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Sat, 11 Sep 2021 14:58:18 -0500
Subject: [PATCH] find headers to copy

---
 Source/Projects/NonWindows/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/Projects/NonWindows/Makefile b/Source/Projects/NonWindows/Makefile
index 1c9e422..c6ca87b 100644
--- a/Source/Projects/NonWindows/Makefile
+++ b/Source/Projects/NonWindows/Makefile
@@ -94,7 +94,7 @@ else
 endif
 
 	mkdir -p $(DESTDIR)$(includedir)/MAC
-	cp $(SRCDIR)/Shared/*.h $(DESTDIR)$(includedir)/MAC
+	cp Shared/*.h $(DESTDIR)$(includedir)/MAC
 
 $(DLLNAME): $(SHAREDOBJECTS) $(LIBOBJECTS) $(DLLOBJECTS)
 	$(CC) $(SHAREDOBJECTS) $(LIBOBJECTS) $(DLLOBJECTS) $(LDOPTS) $(LDFLAGS) $(DLLLDOPTS) -o $@
-- 
2.33.0

