From 79e632660c0af51b3b4a063f4e76c24570d8fe22 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Thu, 29 Apr 2021 14:44:33 -0500
Subject: [PATCH] add-strip-target

---
 Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c2e2715..37d6dfd 100644
--- a/Makefile
+++ b/Makefile
@@ -112,7 +112,7 @@ REGRESS_ARGS	+= "--parse-no-autolink"
 REGRESS_ARGS	+= "--parse-no-cmark"
 REGRESS_ARGS	+= "--parse-no-deflists"
 
-all: lowdown lowdown-diff lowdown.pc
+all: lowdown lowdown-diff lowdown.pc strip
 
 www: $(HTMLS) $(PDFS) $(THUMBS) lowdown.tar.gz lowdown.tar.gz.sha512
 
@@ -130,6 +130,9 @@ lowdown: liblowdown.a main.o
 lowdown-diff: lowdown
 	ln -f lowdown lowdown-diff
 
+strip: lowdown lowdown-diff
+	  strip -x --strip-unneeded lowdown lowdown-diff
+
 liblowdown.a: $(OBJS) $(COMPAT_OBJS)
 	$(AR) rs $@ $(OBJS) $(COMPAT_OBJS)
 
-- 
2.31.1

