diff --unified --recursive --text package.orig/CMakeLists.txt package.new/CMakeLists.txt
--- package.orig/CMakeLists.txt	2022-03-12 15:51:26.653297679 -0600
+++ package.new/CMakeLists.txt	2022-03-12 15:52:05.006630227 -0600
@@ -43,7 +43,7 @@
 find_package(KF5ItemModels ${KF5_WANT_VERSION} CONFIG REQUIRED)
 find_package(KF5PimTextEdit ${KDEPIMTEXTEDIT_VERSION} CONFIG)
 
-set(GPGMEPP_LIB_VERSION "1.13.1")
+set(GPGMEPP_LIB_VERSION "1.17.1")
 find_package(Gpgmepp ${GPGMEPP_LIB_VERSION} CONFIG REQUIRED)
 set_package_properties(Gpgmepp PROPERTIES DESCRIPTION "GpgME++ Library" URL "https://www.gnupg.org" TYPE REQUIRED PURPOSE "GpgME++ is required for OpenPGP support")
 find_package(QGpgme ${GPGMEPP_LIB_VERSION} CONFIG REQUIRED)
