diff --git a/meson.build b/meson.build
index 49bb534..0337626 100644
--- a/meson.build
+++ b/meson.build
@@ -329,7 +329,6 @@ subdir('src/lib/pcre')
 subdir('src/lib/pulse')
 subdir('src/lib/sndio')
 subdir('src/lib/sqlite')
-subdir('src/lib/systemd')
 subdir('src/lib/upnp')
 subdir('src/lib/yajl')
 
@@ -453,7 +452,6 @@ mpd = build_target(
     db_glue_dep,
     storage_glue_dep,
     song_dep,
-    systemd_dep,
     sqlite_dep,
     zeroconf_dep,
     more_deps,
@@ -480,10 +478,6 @@ endif
 
 configure_file(output: 'config.h', configuration: conf)
 
-if systemd_dep.found()
-  subdir('systemd')
-endif
-
 install_data(
   'mpd.svg',
   install_dir: join_paths(get_option('datadir'), 'icons', 'hicolor', 'scalable', 'apps'),
