diff --git a/configure.ac b/configure.ac
index 472c130c22740f32a193868379c3c19d6253d195..6e4baa36b75e652531a9cde33ef446feac82368a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -575,6 +575,7 @@ AC_ARG_ENABLE(pciaccess, AS_HELP_STRING([--enable-pciaccess], [Build Xorg with p
 AC_ARG_ENABLE(linux_acpi, AS_HELP_STRING([--disable-linux-acpi], [Disable building ACPI support on Linux (if available).]), [enable_linux_acpi=$enableval], [enable_linux_acpi=yes])
 AC_ARG_ENABLE(linux_apm, AS_HELP_STRING([--disable-linux-apm], [Disable building APM support on Linux (if available).]), [enable_linux_apm=$enableval], [enable_linux_apm=yes])
 AC_ARG_ENABLE(systemd-logind, AS_HELP_STRING([--enable-systemd-logind], [Build systemd-logind support (default: auto)]), [SYSTEMD_LOGIND=$enableval], [SYSTEMD_LOGIND=auto])
+AC_ARG_ENABLE(elogind, AS_HELP_STRING([--enable-elogind], [Build elogind support (default: auto)]), [ELOGIND_LOGIND=$enableval], [ELOGIND_LOGIND=auto])
 AC_ARG_ENABLE(suid-wrapper, AS_HELP_STRING([--enable-suid-wrapper], [Build suid-root wrapper for legacy driver support on rootless xserver systems (default: no)]), [SUID_WRAPPER=$enableval], [SUID_WRAPPER=no])
 
 dnl DDXes.
@@ -846,6 +847,15 @@ if test "x$WITH_SYSTEMD_DAEMON" = "xyes" -o "x$WITH_SYSTEMD_DAEMON" = "xauto" ;
 fi
 AM_CONDITIONAL([HAVE_SYSTEMD_DAEMON], [test "x$HAVE_SYSTEMD_DAEMON" = "xyes"])
 
+dnl elogind socket fds detection for xtrans
+PKG_CHECK_MODULES(ELOGIND, [libelogind >= 209], [HAVE_ELOGIND=yes], [HAVE_ELOGIND=no])
+if test "x$HAVE_ELOGIND" = xyes -a "x$HAVE_SYSTEMD_DAEMON" = xno; then
+        REQUIRED_LIBS="$REQUIRED_LIBS libelogind >= 209"
+else
+        HAVE_ELOGIND="no"
+fi
+AM_CONDITIONAL([HAVE_ELOGIND], [test "x$HAVE_ELOGIND" = "xyes"])
+
 if test "x$CONFIG_UDEV" = xyes && test "x$CONFIG_HAL" = xyes; then
 	AC_MSG_ERROR([Hotplugging through both libudev and hal not allowed])
 fi
@@ -878,6 +888,7 @@ if test "x$CONFIG_UDEV" = xyes; then
 fi
 AM_CONDITIONAL(CONFIG_UDEV_KMS, [test "x$CONFIG_UDEV_KMS" = xyes])
 
+
 PKG_CHECK_MODULES(DBUS, $LIBDBUS, [HAVE_DBUS=yes], [HAVE_DBUS=no])
 if test "x$HAVE_DBUS" = xyes; then
 	AC_DEFINE(HAVE_DBUS, 1, [Have D-Bus support])
@@ -918,6 +929,27 @@ if test "x$SYSTEMD_LOGIND" = xyes; then
 fi
 AM_CONDITIONAL(SYSTEMD_LOGIND, [test "x$SYSTEMD_LOGIND" = xyes])
 
+dnl Chcek for elogind integration
+if test "x$ELOGIND_LOGIND" = xauto; then
+        if test "x$HAVE_DBUS" = xyes -a "x$CONFIG_UDEV" = xyes -a "x$HAVE_ELOGIND" = xyes -a "x$SYSTEMD_LOGIND" = xno; then
+                ELOGIND_LOGIND=yes
+        else
+                ELOGIND_LOGIND=no
+        fi
+fi
+if test "x$ELOGIND_LOGIND" = xyes; then
+        if ! test "x$HAVE_DBUS" = xyes; then
+                AC_MSG_ERROR([elogind requested, but D-Bus is not installed.])
+        fi
+        if ! test "x$CONFIG_UDEV" = xyes ; then
+                AC_MSG_ERROR([elogind is only supported in combination with udev configuration.])
+        fi
+
+        AC_DEFINE(ELOGIND_LOGIND, 1, [Enable elogind integration])
+        NEED_DBUS="yes"
+fi
+AM_CONDITIONAL(ELOGIND_LOGIND, [test "x$ELOGIND_LOGIND" = xyes])
+
 if test "x$SUID_WRAPPER" = xyes; then
         dnl This is a define so that if some platforms want to put the wrapper
         dnl somewhere else this can be easily changed
diff --git a/hw/kdrive/src/kdrive.c b/hw/kdrive/src/kdrive.c
index 77114dc46f73ed50cc1180dba624a9e7393bed08..661bd46553e6d70eb0c2138a1fa41acfa965c736 100644
--- a/hw/kdrive/src/kdrive.c
+++ b/hw/kdrive/src/kdrive.c
@@ -1001,7 +1001,7 @@ xf86_find_platform_device_by_devnum(int major, int minor)
     return NULL;
 }
 
-#ifdef SYSTEMD_LOGIND
+#if defined (SYSTEMD_LOGIND) || defined(ELOGIND_LOGIND)
 void
 systemd_logind_vtenter(void)
 {
diff --git a/hw/xfree86/os-support/linux/Makefile.am b/hw/xfree86/os-support/linux/Makefile.am
index 7d4cf4222fa25a20a8a3de045a450e95c390cad6..e4f11155249e6a01145b85088f17a26bbf0ebea1 100644
--- a/hw/xfree86/os-support/linux/Makefile.am
+++ b/hw/xfree86/os-support/linux/Makefile.am
@@ -24,6 +24,11 @@ LOGIND_SRCS = systemd-logind.c
 XORG_CFLAGS += $(DBUS_CFLAGS)
 endif
 
+if ELOGIND_LOGIND
+LOGIND_SRCS = systemd-logind.c
+XORG_CFLAGS += $(DBUS_CFLAGS)
+endif
+
 liblinux_la_SOURCES = linux.h lnx_init.c lnx_video.c \
                      lnx_agp.c lnx_kmod.c lnx_bell.c lnx_platform.c \
 		     $(srcdir)/../shared/VTsw_usl.c \
diff --git a/include/dix-config.h.in b/include/dix-config.h.in
index 855b3d50c11ef603a3ffd7267576bc2ca866a2c8..57f3c784e196008b0bac1e9a3fa0eac1db9ab0dd 100644
--- a/include/dix-config.h.in
+++ b/include/dix-config.h.in
@@ -418,6 +418,9 @@
 /* Enable systemd-logind integration */
 #undef SYSTEMD_LOGIND 1
 
+/* Enable elogind integration */
+#undef ELOGIND_LOGIND 1
+
 /* Have a monotonic clock from clock_gettime() */
 #undef MONOTONIC_CLOCK
 
diff --git a/include/meson.build b/include/meson.build
index 04c41e9992a7ab48486ccfb149b91b7c83d7b722..88d4d4dc2f93f985cc4ca757661e8b7d827f9850 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -88,6 +88,7 @@ conf_data.set('CONFIG_UDEV_KMS', build_udev)
 conf_data.set('HAVE_DBUS', build_dbus)
 conf_data.set('CONFIG_HAL', build_hal)
 conf_data.set('SYSTEMD_LOGIND', build_systemd_logind)
+conf_data.set('ELOGIND_LOGIND', build_systemd_logind)
 conf_data.set('NEED_DBUS', build_systemd_logind or build_hal)
 conf_data.set('CONFIG_WSCONS', host_machine.system() == 'openbsd')
 
diff --git a/include/systemd-logind.h b/include/systemd-logind.h
index a4067d097547ce49da385e63c51209f32ef1aa9f..aed7d50f6421ae2a46141b0ba7fd16023bee544a 100644
--- a/include/systemd-logind.h
+++ b/include/systemd-logind.h
@@ -26,7 +26,7 @@
 #ifndef SYSTEMD_LOGIND_H
 #define SYSTEMD_LOGIND_H
 
-#ifdef SYSTEMD_LOGIND
+#if defined(SYSTEMD_LOGIND) || defined(ELOGIND_LOGIND)
 int systemd_logind_init(void);
 void systemd_logind_fini(void);
 int systemd_logind_take_fd(int major, int minor, const char *path, Bool *paus);
