diff --git a/src/upnp/protocol/pupnp/pupnp.cpp b/src/upnp/protocol/pupnp/pupnp.cpp
index 7652c2e3c..e097085da 100644
--- a/src/upnp/protocol/pupnp/pupnp.cpp
+++ b/src/upnp/protocol/pupnp/pupnp.cpp
@@ -85,16 +85,12 @@ PUPnP::PUPnP()
 {
     int upnp_err = UPNP_E_SUCCESS;
 
-#if UPNP_ENABLE_IPV6
     upnp_err = UpnpInit2(0, 0);
     if (upnp_err != UPNP_E_SUCCESS) {
         JAMI_WARN("PUPnP: UpnpInit2 Failed to initialize");
         UpnpFinish();					// Destroy threads before reusing upnp init function.
         upnp_err = UpnpInit(0, 0);      // Deprecated function but fall back on it if UpnpInit2 fails.
     }
-#else
-    upnp_err = UpnpInit(0, 0);           // Deprecated function but fall back on it if IPv6 not enabled.
-#endif
 
     if (upnp_err != UPNP_E_SUCCESS) {
         JAMI_ERR("PUPnP: Can't initialize libupnp: %s", UpnpGetErrorMessage(upnp_err));
diff --git a/src/upnp/protocol/pupnp/pupnp.cpp b/src/upnp/protocol/pupnp/pupnp.cpp
index e097085da..3a620a0ec 100644
--- a/src/upnp/protocol/pupnp/pupnp.cpp
+++ b/src/upnp/protocol/pupnp/pupnp.cpp
@@ -89,7 +89,6 @@ PUPnP::PUPnP()
     if (upnp_err != UPNP_E_SUCCESS) {
         JAMI_WARN("PUPnP: UpnpInit2 Failed to initialize");
         UpnpFinish();					// Destroy threads before reusing upnp init function.
-        upnp_err = UpnpInit(0, 0);      // Deprecated function but fall back on it if UpnpInit2 fails.
     }
 
     if (upnp_err != UPNP_E_SUCCESS) {
