From 48d795a9c1ad4c77466878eb58a81567c3e16138 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Mon, 24 Jan 2022 22:37:49 -0600
Subject: [PATCH] ffmpeg-code-makefile_am

---
 modules/codec/Makefile.am | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/modules/codec/Makefile.am b/modules/codec/Makefile.am
index 3dadf11..3ab9537 100644
--- a/modules/codec/Makefile.am
+++ b/modules/codec/Makefile.am
@@ -414,12 +414,9 @@ libvaapi_plugin_la_SOURCES = \
 libvaapi_plugin_la_CPPFLAGS = $(AM_CPPFLAGS)
 libvaapi_plugin_la_CFLAGS = $(AM_CFLAGS) $(AVCODEC_CFLAGS)
 libvaapi_plugin_la_LIBADD = $(LIBVA_LIBS)
-if HAVE_AVCODEC_VAAPI
 if HAVE_VAAPI_DRM
 codec_LTLIBRARIES += libvaapi_drm_plugin.la
 endif
-codec_LTLIBRARIES += libvaapi_plugin.la
-endif
 
 libd3d9_common_la_SOURCES = video_chroma/d3d9_fmt.c video_chroma/d3d9_fmt.h \
 	video_chroma/dxgi_fmt.c video_chroma/dxgi_fmt.h
-- 
2.35.0

