From a180dee757f54ff228f970b5bef4953069cf0e12 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Sun, 29 Aug 2021 10:03:58 -0500
Subject: [PATCH] fix format security

---
 converter/ppm/ximtoppm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/converter/ppm/ximtoppm.c b/converter/ppm/ximtoppm.c
index 527b25f..1e0cff8 100644
--- a/converter/ppm/ximtoppm.c
+++ b/converter/ppm/ximtoppm.c
@@ -116,9 +116,9 @@ ReadXimHeader(FILE *     const in_fp,
 */
     header->bits_channel = atoi(a_head.bits_per_channel);
     header->alpha_flag = atoi(a_head.alpha_channel);
-    pm_asprintf(&header->author,  a_head.author);
-    pm_asprintf(&header->date,    a_head.date);
-    pm_asprintf(&header->program, a_head.program);
+    pm_asprintf(&header->author, "%s",  a_head.author);
+    pm_asprintf(&header->date, "%s",   a_head.date);
+    pm_asprintf(&header->program, "%s", a_head.program);
     /* Do double checking for bakwards compatibility */
     if (header->npics == 0)
         header->npics = 1;
-- 
2.33.0

