From 6340278a3e708a5aface47b7aa0efdc77f687111 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens@artixlinux.org>
Date: Wed, 10 Nov 2021 16:29:41 -0600
Subject: [PATCH] Correct tbb cmake path

---
 cmake/OpenCVDetectTBB.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/OpenCVDetectTBB.cmake b/cmake/OpenCVDetectTBB.cmake
index fc564e9..a33469d 100644
--- a/cmake/OpenCVDetectTBB.cmake
+++ b/cmake/OpenCVDetectTBB.cmake
@@ -19,7 +19,7 @@
 #   - "tbb" target exists and added to OPENCV_LINKER_LIBS
 
 function(ocv_tbb_cmake_guess _found)
-    find_package(TBB QUIET COMPONENTS tbb PATHS "$ENV{TBBROOT}/cmake" "$ENV{TBBROOT}/lib/cmake/tbb")
+    find_package(TBB QUIET COMPONENTS tbb PATHS "$ENV{TBBROOT}/cmake" "$ENV{TBBROOT}/lib/cmake/TBB")
     if(TBB_FOUND)
       if(NOT TARGET TBB::tbb)
         message(WARNING "No TBB::tbb target found!")
-- 
2.33.1

