 bin/pkg/comparepkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/pkg/comparepkg.in b/bin/pkg/comparepkg.in
index 50119ee..f68a919 100644
--- a/bin/pkg/comparepkg.in
+++ b/bin/pkg/comparepkg.in
@@ -54,7 +54,7 @@ compare_u(){
     if [ "$result" -eq -1 ];then
 #         local arel=${archver##*-} brel=${artixver##*-}
 #         local reldiff
-        reldiff=$(( $arel-${brel%.*} ))
+#         reldiff=$(( $arel-${brel%.*} ))
         msg_row_red "${table}" "${a}" "${b}" "$pkg" "$archver" "$artixver" "${group#*-}"
         write_list "${a}:$pkg" pkg_upgrades
     fi
