diff -Naur cjs-5.2.0-orig/installed-tests/js/meson.build cjs-5.2.0/installed-tests/js/meson.build
--- cjs-5.2.0-orig/installed-tests/js/meson.build	2021-11-15 08:51:30.000000000 -0300
+++ cjs-5.2.0/installed-tests/js/meson.build	2021-12-17 16:45:47.076528128 -0300
@@ -48,7 +48,7 @@
     sources: regress_sources, namespace: 'Regress', nsversion: '1.0',
     identifier_prefix: 'Regress', symbol_prefix: 'regress_',
     extra_args: ['--warn-all', '--warn-error'] + regress_gir_c_args,
-    install: get_option('installed_tests'), install_dir_gir: false,
+    install: get_option('installed_tests'), install_dir_gir: [false],
     install_dir_typelib: installed_tests_execdir)
 regress_typelib = regress_gir[1]
 
@@ -66,7 +66,7 @@
 warnlib_gir = gnome.generate_gir(libwarnlib, includes: ['Gio-2.0'],
     sources: warnlib_sources, namespace: 'WarnLib', nsversion: '1.0',
     symbol_prefix: 'warnlib_', header: 'warnlib.h',
-    install: get_option('installed_tests'), install_dir_gir: false,
+    install: get_option('installed_tests'), install_dir_gir: [false],
     install_dir_typelib: installed_tests_execdir)
 warnlib_typelib = warnlib_gir[1]
 
@@ -82,7 +82,7 @@
     includes: ['Gio-2.0'], sources: gimarshallingtests_sources,
     namespace: 'GIMarshallingTests', nsversion: '1.0',
     symbol_prefix: 'gi_marshalling_tests_', extra_args: '--warn-error',
-    install: get_option('installed_tests'), install_dir_gir: false,
+    install: get_option('installed_tests'), install_dir_gir: [false],
     install_dir_typelib: installed_tests_execdir)
 gimarshallingtests_typelib = gimarshallingtests_gir[1]
 
diff -Naur cjs-5.2.0-orig/meson.build cjs-5.2.0/meson.build
--- cjs-5.2.0-orig/meson.build	2021-11-15 08:51:30.000000000 -0300
+++ cjs-5.2.0/meson.build	2021-12-17 16:46:09.869860858 -0300
@@ -542,7 +542,7 @@
     includes: ['GObject-2.0', 'Gio-2.0'], sources: libgjs_private_sources,
     namespace: 'CjsPrivate', nsversion: '1.0', identifier_prefix: 'Gjs',
     symbol_prefix: 'gjs_', extra_args: '--warn-error', install: true,
-    install_dir_gir: false, install_dir_typelib: pkglibdir / 'girepository-1.0')
+    install_dir_gir: [false], install_dir_typelib: pkglibdir / 'girepository-1.0')
 gjs_private_typelib = gjs_private_gir[1]
 
 ### Build gjs-console interpreter ##############################################
