Index: src/c128/z80.c
===================================================================
--- src/c128/z80.c	(revision 37526)
+++ src/c128/z80.c	(working copy)
@@ -444,7 +444,7 @@
 
 /* ------------------------------------------------------------------------- */
 
-z80_regs_t z80_regs;
+extern z80_regs_t z80_regs;
 
 static void import_registers(void)
 {
Index: src/rs232drv/rs232.h
===================================================================
--- src/rs232drv/rs232.h	(revision 37526)
+++ src/rs232drv/rs232.h	(working copy)
@@ -78,7 +78,7 @@
 #define RS232_NUM_DEVICES 4
 
 extern char *rs232_devfile[RS232_NUM_DEVICES];
-int rs232_useip232[RS232_NUM_DEVICES];
+extern int rs232_useip232[RS232_NUM_DEVICES];
 
 /* the "ip232" protocol used by tcpser 
  
