--- ../usr/iscsid.c	2019-09-17 00:15:43.182127419 +0200
+++ ../usr/iscsid.c	2019-09-17 00:17:17.816563188 +0200
@@ -34,9 +34,6 @@
 #include <sys/wait.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#ifndef	NO_SYSTEMD
-#include <systemd/sd-daemon.h>
-#endif

 #include "iscsid.h"
 #include "mgmt_ipc.h"
@@ -341,25 +338,13 @@
 /* called right before we enter the event loop */
 static void set_state_to_ready(void)
 {
-#ifndef	NO_SYSTEMD
-	if (sessions_to_recover)
-		sd_notify(0, "READY=1\n"
-				"RELOADING=1\n"
-				"STATUS=Syncing existing session(s)\n");
-	else
-		sd_notify(0, "READY=1\n"
-				"STATUS=Ready to process requests\n");
-#endif
+
 }

 /* called when recovery process has been reaped */
 static void set_state_done_reloading(void)
 {
-#ifndef	NO_SYSTEMD
-	sessions_to_recover = 0;
-	sd_notifyf(0, "READY=1\n"
-			"STATUS=Ready to process requests\n");
-#endif
+
 }

 int main(int argc, char *argv[])
