--- calamares-3.2.18/src/modules/machineid/main.py	2020-01-28 18:42:21.000000000 +0200
+++ calamares/src/modules/machineid/main.py	2020-02-07 11:59:06.872485260 +0200
@@ -4,7 +4,7 @@
 # === This file is part of Calamares - <https://github.com/calamares> ===
 #
 #   Copyright 2014, Kevin Kofler <kevin.kofler@chello.at>
-#   Copyright 2016, Philip Müller <philm@manjaro.org>
+#   Copyright 2016,2020 Philip Müller <philm@manjaro.org>
 #   Copyright 2017, Alf Gaida <agaida@siduction.org>
 #   Copyright 2019, Adriaan de Groot <groot@kde.org>
 #
@@ -54,6 +54,10 @@
     enable_symlink = libcalamares.job.configuration["symlink"]
     target_systemd_machineid_file = root_mount_point + "/etc/machine-id"
     target_dbus_machineid_file = root_mount_point + "/var/lib/dbus/machine-id"
+    target_dbus_folder = root_mount_point + "/var/lib/dbus"
+    
+    if not os.path.exists(target_dbus_folder):
+        os.mkdir(target_dbus_folder, 0644)
 
     if os.path.exists(target_dbus_machineid_file):
         os.remove(target_dbus_machineid_file)
@@ -65,7 +69,7 @@
 
     if enable_dbus:
         if enable_symlink and os.path.exists(target_systemd_machineid_file):
-            check_target_env_call(["ln", "-s", "/etc/machine-id",
+            check_target_env_call(["ln", "-sf", "/etc/machine-id",
                                    "/var/lib/dbus/machine-id"])
         else:
             check_target_env_call(["dbus-uuidgen", "--ensure"])
